### ISSUE TITLE
Acceptable for a module to use gToolbar (and change cable tension/opacity)?

### ISSUE THREAD

**Issue by alikins** [VCVRack/Rack#1056]
I wrote a module that changes application wide cable tension/opacity settings via gToolbar and was wondering if that is acceptable for a module released via plugin manager?

The code: https://github.com/alikins/AlikinsLab-rack-plugins/blob/master/src/CableMadness.cpp
Facebook discussion: https://www.facebook.com/groups/vcvrack/permalink/210196246307194/

My gut feeling is it is probably a misuse of the API but figured I'd ask. I wouldn't expect the details of gToolbar to remain API compatible across 0.6.x for one example.  Also not sure if it's really the kind of thing a module should be doing, but since some folks seemed interested in it I wanted to confirm.

**Comment by AndrewBelt** [comment]
My policy is

>Nothing is hard-encapsulated in Rack. You are free to hack anything you wish in a plugin. Instead, I believe in soft-deprecation. Once the API is stabalized in Rack v1, I will make each private symbol clear through docstrings.

In particular, cable tension/opacity will be moved to another header and privatized, so it *will* break in Rack v1. However, if you're still around to change things, you should be able to chase this private symbol around to fix your plugin.

**Comment by MarcBoule** [comment]
Is it the same regarding programmatically unplugging a wire in one of our modules, i.e. the mechanism might change but by changing our approach we will still be able to do it?

**Comment by AndrewBelt** [comment]
Yes, that is likely among the functionality that will be privatized.

**Comment by MarcBoule** [comment]
Thanks! If I can ask one more thing regarding 1.0: I understand that in our code we cannot assume there will always be a ModuleWidget for our Module, but am I correct in assuming there will always be the Widgets themselves (i.d. SVGKnob, SVGSwitch, and such)?

**Comment by AndrewBelt** [comment]
The childen widgets are created in the ModuleWidget subclass's constructor.
